js/app/Visualization/UI/BasicSidebar.js

```diff
@@ -176,8 +176,39 @@ define([
             self.node.find(".vessel_id .flag").html("---");
           }
 
-          self.node.find(".vessel_id .imo").html(data.imo || "---");
-          self.node.find(".vessel_id .mmsi").html(data.mmsi || "---");
+          if (data.imo) {
```

Comment:
@redhog Both imo and mmsi code are basically the same. Why don't you encapsulate them?

display_with_links(data.imo, ".vessel_id .imo", "link")

```javascript
display_with_link(values, css_style, the_url){
          if (values) {
             self.node.find(css_style).html("");
             var first = true;
             values.split(",").map(function (value) {
               var link = $("<a target='_blank'>");
               link.text(mmsi);
               link.attr({href: the_url + value});
               if (!first) {
                 self.node.find(css_style).append(", ");
               }
               self.node.find(css_style).append(link);
               first = false;
             });
           } else {
             self.node.find(css_style).html("---");
  }
}
```